### PR TITLE
Fix happy path

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: frontend/src/Maker.tsx [hunk 1]
```diff
@@ -31,7 +31,14 @@ interface CfdSellOrderPayload {
 }
 
 async function postCfdSellOrderRequest(payload: CfdSellOrderPayload) {
-    let res = await fetch(`/api/order/sell`, { method: "POST", body: JSON.stringify(payload), credentials: "include" });
+    let res = await fetch(`/api/order/sell`, {
+        method: "POST",
+        body: JSON.stringify(payload),
+        headers: {
+            "Content-Type": "application/json",
+        },
```
Comment (klochowicz) [inline frontend/src/Maker.tsx:39]: out of curiosity: would this header be needed if we stayed with `axiom` after adding credentials? 
(I'm trying to figure out the pros and cons of each)

**Comment by klochowicz** [comment]
When trying to establish a CFD, I still run into the issue I mentioned last night in my PR: 

```
thread 'rocket-worker-thread' panicked at 'Sorry, have to panic 😬 - the amounts that the counterparty sent were wrong, expected 0.00004000 BTC actual 0.00040000 BTC', daemon/src/setup_contract_actor.rs:57:13
```

**Review by klochowicz — APPROVED** [review]
these are useful changes, and it would be great to get them in asap due to upcoming taker refactor, but I'm not sure if the changeset completely fixes the happy path as advertised (see my comment)

**Comment by thomaseizinger** [comment]
> When trying to establish a CFD, I still run into the issue I mentioned last night in my PR: 
> 
> ```
> thread 'rocket-worker-thread' panicked at 'Sorry, have to panic 😬 - the amounts that the counterparty sent were wrong, expected 0.00004000 BTC actual 0.00040000 BTC', daemon/src/setup_contract_actor.rs:57:13
> ```

Huh, it doesnt do that for me!

**Comment by klochowicz** [comment]
> > When trying to establish a CFD, I still run into the issue I mentioned last night in my PR:
> > ```
> > thread 'rocket-worker-thread' panicked at 'Sorry, have to panic 😬 - the amounts that the counterparty sent were wrong, expected 0.00004000 BTC actual 0.00040000 BTC', daemon/src/setup_contract_actor.rs:57:13
> > ```
> 
> Huh, it doesnt do that for me!

Does the CFD get to the 'open' state on both ends? I tried following @da-kami suggestion by using his seed files (with some funds) and trying to start a CFD worth $2.  I had the same thing yesterday, so it's not related to your changes - I was just hoping that "fixing the happy path" would also fix that ;D

**Comment by thomaseizinger** [comment]
Yes I get into CFD open in both cases, check the screenshots in the simplify UI PR!

**Comment by da-kami** [comment]
> Yes I get into CFD open in both cases, check the screenshots in the simplify UI PR!

`PendingOpen` to be precise - which  is correct :)

**Comment by thomaseizinger** [comment]
@da-kami / @klochowicz Please merge if you are happy with this!

**Review by da-kami — APPROVED** [review]
I had a look and it seems somewhat reasonable but I don't have time to test it on my machine at the moment.
I am surprised we need the buffer on the taker side - it was not needed before, but might as well just add it. 

Feel free to merge, I will see to test later to confirm it works. If it does not we can go on debugging on top of this work.

**Comment by thomaseizinger** [comment]
>  I am surprised we need the buffer on the taker side - it was not needed before, but might as well just add it.

We didn't need it because the maker was waiting for a pointless `SetupContract` message and hence always delayed in its setup.

**Pull request merged**